### DISCUSSION TITLE
Cannot get PyVista to display the correct topology

### DISCUSSION THREAD

**Discussion by MishalMohanlalZut** [pyvista/pyvista#6323] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import pyvista as pv
import numpy as np

def pvplotR(Nodes, Connectivity, Result, Dfactor):
    # Create initial points array from node positions
    points = Nodes[:, 1:4]
    node_labels = Nodes[:, 1:4].astype(int)

    plotter = pv.Plotter()
    
    for t in range(Result.shape[1]):
        # Update points using displacement data
        displacements = Result[:3*len(Nodes), t].reshape((len(Nodes), 3))
        new_points = points + displacements * Dfactor
        print(new_points)
        displacement_magnitude = np.linalg.norm(displacements, axis=1)
        
        # Create a PolyData object
        mesh = pv.PolyData(new_points)
        
        # Create and set lines
        lines = np.hstack([[2, row[1], row[2]] for row in Connectivity])
        mesh.lines = lines
        print(lines)
        mesh["Displacement Magnitude"] = displacement_magnitude
        
        # Save the mesh to a .vtk file
        filename = f"time_step_{t:03d}.vtk"
        mesh.save(filename)
        print(f"Saved {filename}")
        
        plotter.clear()  # Clear the plotter for each time step
        plotter.add_mesh(mesh, scalars="Displacement Magnitude", cmap="coolwarm", show_scalar_bar=True, line_width=4)
        plotter.add_point_labels(new_points, node_labels, point_size=20, text_color='black', font_size=12, name='labels')
    
    plotter.show()

# Example usage
Nodes = np.array([
    [1, 0, 0, 0],
    [2, 0, 197, 0],
    [3, 132, 197, 0],
    [4, 132, 0, 0],
    [5, 0, 63, 0],
    [6, 132, 63, 0]
])  # Example nodes
Connectivity = np.array([
    [2, 1, 5],
    [2, 2, 3],
    [2, 3, 6],
    [2, 5, 2],
    [2, 6, 4],
    [2, 5, 6],
    [2, 2, 6],
    [2, 3, 5]
])  # Example connectivity
Result = np.random.rand(3*len(Nodes), 10)*0  # Example result with 10 timesteps
Dfactor = 0.1  # Example displacement factor

pvplotR(Nodes, Connectivity, Result, Dfactor)
```

### Description

I have written the pvplotR function which takes results from a transient dynamic analysis which I have performed on Python and attempts to plot the data. The resulting plot should lines with a color contour however the resulting plot is not correct. I have checked the topology and the nodes which are all correct and inline with the expectations of what PyVista requires. Can someone please assist with an explanation of what I have done incorrect?

### System Information

```shell
Date: Mon Jul 01 09:40:48 2024 South Africa Standard Time

                OS : Windows (10 10.0.19045 SP0 Multiprocessor Free)
            CPU(s) : 20
           Machine : AMD64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA RTX A1000 Laptop GPU/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 528.79
  MathText Support : False

  Python 3.11.3 | packaged by conda-forge | (main, Apr  6 2023, 08:50:54) [MSC
  v.1934 64 bit (AMD64)]

           pyvista : 0.43.10
               vtk : 9.3.0
             numpy : 1.26.4
        matplotlib : 3.8.4
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.3.0
             PyQt5 : 5.15.9
             scipy : 1.13.1
--------------------------------------------------------------------------------
```

### Screenshots

This is what is being plotted
![image](https://github.com/pyvista/pyvista/assets/146818267/4d0892f8-0c69-42c0-ba67-6158b4238a6b)

The image shape should look like this
![image](https://github.com/pyvista/pyvista/assets/146818267/a24b1d69-1e6b-4074-a137-24f9f1cc1fb9)

**Comment by MatthewFlamm** [comment, marked as answer]
Python is 0 indexed yet your Connectivity array looks to be 1 indexed.  I'm surprised there is no segfault here.

**Reply by MishalMohanlalZut** [reply]
Aaah thank you!!! Rookie mistake from a Julia and Matlab user
